spock-core/src/main/java/org/spockframework/runtime/extension/builtin/StepwiseExtension.java

```diff
@@ -18,21 +18,45 @@
 import org.spockframework.runtime.extension.IAnnotationDrivenExtension;
 import org.spockframework.runtime.model.*;
 import org.spockframework.runtime.model.parallel.ExecutionMode;
+import spock.lang.Stepwise;
 
-import java.lang.annotation.Annotation;
 import java.util.List;
 
-public class StepwiseExtension implements IAnnotationDrivenExtension {
+public class StepwiseExtension implements IAnnotationDrivenExtension<Stepwise> {
   @Override
-  public void visitSpecAnnotation(Annotation annotation, final SpecInfo spec) {
+  public void visitSpecAnnotation(Stepwise annotation, final SpecInfo spec) {
     sortFeaturesInDeclarationOrder(spec);
     includeFeaturesBeforeLastIncludedFeature(spec);
     skipFeaturesAfterFirstFailingFeature(spec);
 
-    // Disable parallel child execution for @Stepwise tests
+    // Disable parallel child execution for @Stepwise specs
     spec.setChildExecutionMode(ExecutionMode.SAME_THREAD);
   }
 
+  @Override
+  public void visitFeatureAnnotation(Stepwise annotation, FeatureInfo feature) {
+    // Disable parallel iteration execution for @Stepwise features
+    feature.setExecutionMode(ExecutionMode.SAME_THREAD);
+
+    // If an error occurs in this feature, skip remaining iterations
+    feature.getSpec().addListener(new FeatureSkippingErrorListener(feature));
+  }
+
+  private static class FeatureSkippingErrorListener extends AbstractRunListener {
```

Comment:
As discussed in our chat I would prefer using a targeted interceptor (`feature.featureMethod.addInterceptor`), so that we don't add multiple listener to the whole spec.
```groovy
intercept(invocation) {
  try {
    invocation.proceed()
   }
   catch (Throwable e) {
     invocation.feature.skip()
     throw e
   }
}
```

Reply:
Sure, I can do that.

But for the protocol: An error listener is only called when an arror occurs, which should be a rare case. A feature method interceptor is always called, i.e. it has to do its "try, catch, proceed" thing for every iteration of the feature it is registered on. Let's say there are 10 iterations, i.e. it is called 10x for every test execution of that feature method. The error listener however, even though it listens for errors in the whole spec, is only called if an error occurs. Let us be generous and say that 1 test fails in 20% of test runs (which would be really flaky), i.e. per test run the error listener is called 0.2x on average. Compare that to 10x. It would be a factor of 50. Even if 1 random test fails during every single test run (what a nightmare!), it is still a factor of 10.

So the reason why I am following your advice is not the efficiency or performance which you seem to have in mind, but the advantage that registering an interceptor on the feature does not need to keep a reference to same feature like the error listener does. It just looks nicer in code, even though it leads to more overhead in many use cases. If you are fine with that, so am I.